services/payment_cancel/lambda_function.py

```diff
@@ -0,0 +1,72 @@
+import json
+import os
+
+import boto3
+
+ORDERS_TABLE = os.environ['ORDERS_TABLE']
+USERS_TABLE = os.environ['USERS_TABLE']
+
+dynamodb = boto3.resource('dynamodb')
+orders_table = dynamodb.Table(ORDERS_TABLE)
+
+
+def get_order(order_id):
+    """Obtain the order by the given id."""
+    response = orders_table.get_item(Key={'id': order_id})
+    return response['Item']
+
+
+def revert_order(order):
+    """Revert the specified order and add the costs back to the user's credit."""
+    dynamodb.meta.client.transact_write_items(
+        TransactItems=[
+            {
+                'Update': {
+                    'TableName': ORDERS_TABLE,
+                    'Key': {'id': order['id']},
+                    'UpdateExpression': 'SET paid = :paid',
+                    'ConditionExpression': 'paid = :previousPaid',
+                    'ExpressionAttributeValues': {
+                        ':paid': False,
+                        ':previousPaid': True,
+                    },
+                },
+
+            },
+            {
+                'Update': {
+                    'TableName': USERS_TABLE,
+                    'Key': {'id': order['user_id']},
+                    'UpdateExpression': 'ADD credit :cost',
```

Comment:
same thing, in the documentation it's advised to use SET for these types of operations

Reply:
Where did you find this? I only see a mention that in general they advise you to use `SET`.

Reply:
You can read more about this here: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.UpdateExpressions.html
Please change the ADD to SET !

Reply:
Yep, I have seen that, but that does not explain why and when we should (not) use `ADD`. It is just a general recommendation. Is there a strong reason for not using `ADD`?

Reply:
There are a couple reasons for why it's better to use SET in this case, one of them being that we should be consistent with what we have already use for this project and in all other places we used SET. I would like to keep it like that. Thanks